salt/transport/zeromq.py

```diff
@@ -118,242 +98,23 @@ def _get_master_uri(master_ip, master_port, source_ip=None, source_port=None):
-            data = yield self.message_client.send(
-                self._package_load(self.auth.crypticle.dumps(load)),
-                timeout=timeout,
-                tries=tries,
-            )
-            # we may not have always data
-            # as for example for saltcall ret submission, this is a blind
-            # communication, we do not subscribe to return events, we just
-            # upload the results to the master
-            if data:
-                data = self.auth.crypticle.loads(data, raw)
-            if not raw:
-                data = salt.transport.frame.decode_embedded_strs(data)
-            raise salt.ext.tornado.gen.Return(data)
-
-        if not self.auth.authenticated:
-            # Return control back to the caller, resume when authentication succeeds
-            yield self.auth.authenticate()
-        try:
-            # We did not get data back the first time. Retry.
-            ret = yield _do_transfer()
-        except salt.crypt.AuthenticationError:
-            # If auth error, return control back to the caller, continue when authentication succeeds
-            yield self.auth.authenticate()
-            ret = yield _do_transfer()
-        raise salt.ext.tornado.gen.Return(ret)
-
-    @salt.ext.tornado.gen.coroutine
-    def _uncrypted_transfer(self, load, tries=3, timeout=60):
-        """
-        Send a load across the wire in cleartext
-
-        :param dict load: A load to send across the wire
-        :param int tries: The number of times to make before failure
-        :param int timeout: The number of seconds on a response before failing
-        """
-        ret = yield self.message_client.send(
-            self._package_load(load),
-            timeout=timeout,
-            tries=tries,
-        )
-
-        raise salt.ext.tornado.gen.Return(ret)
-
-    @salt.ext.tornado.gen.coroutine
-    def send(self, load, tries=3, timeout=60, raw=False):
-        """
-        Send a request, return a future which will complete when we send the message
-        """
-        if self.crypt == "clear":
-            ret = yield self._uncrypted_transfer(load, tries=tries, timeout=timeout)
-        else:
-            ret = yield self._crypted_transfer(
-                load, tries=tries, timeout=timeout, raw=raw
-            )
-        raise salt.ext.tornado.gen.Return(ret)
-
-
-class AsyncZeroMQPubChannel(
-    salt.transport.mixins.auth.AESPubClientMixin, salt.transport.client.AsyncPubChannel
-):
+class PublishClient(salt.transport.base.PublishClient):
     """
     A transport channel backed by ZeroMQ for a Salt Publisher to use to
     publish commands to connected minions
     """
 
-    async_methods = [
-        "connect",
-        "_decode_messages",
-    ]
-    close_methods = [
-        "close",
-    ]
+    ttype = "zeromq"
 
-    def __init__(self, opts, **kwargs):
+    def __init__(self, opts, io_loop, **kwargs):  # pylint: disable=W0231
```

Comment:
**question** why do we want to disable `super-init-not-called`? Is this another one of those places that we'll need some further cleanup?

Reply:
This was just because the parent class's __init__ is a no-op. I've changed this to call the no-op so we don't need to ignore anything in the linter.